code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -95,7 +95,7 @@ protected void setName(Instance instance, Map<Object, Object> data) {
         if (instance.getName() != null) {
             return;
         }
-        String name = "r-" + instance.getUuid().substring(0, 4);
+        String name = "r-" + instance.getUuid().substring(0, 10);
```

Comment:
@cjellick unrelated, forgot to mention - we generate container name if passed as null via API. Increased the size that we strip off the UUID to reduce potential chances for duplicates.